README.md

```diff
@@ -1 +1,171 @@
-# hello-react-front-end
\ No newline at end of file
+# Hello rails react
+
+<a name="readme-top"></a>
+
+# 📗 Table of Contents
+
+- [📖 About the Project](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+- [💻 Getting Started](#getting-started)
+  - [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+  - [Install](#install)
+  - [Usage](#usage)
+  - [Run tests](#run-tests)
+- [👥 Authors](#authors)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [❓ FAQ](#faq)
+- [📝 License](#license)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 [Hello rails react] <a name="about-project"></a>
+
+**[Hello rails react]** is a simple app Backend with Rails and Frontend with React, that generates random greetings.
+
+## Link to Backend App: https://github.com/TayyabKM/hello-rails-back-end
+
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+
+<details>
+  <summary>Ruby on Rails</summary>
+  <ul>
+    <li><a>https://rubyonrails.org/</a></li>
+  </ul>
+  <summary>React</summary>
+  <ul>
+    <li><a>https://react.dev/</a></li>
+  </ul>
+  <summary>Redux</summary>
+  <ul>
+    <li><a>https://redux.js.org/</a></li>
+  </ul>
+</details>
+
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+- **[Generate random Greetings]**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- GETTING STARTED -->
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+To get a local copy up and running, follow these steps.
+
+### Prerequisites
+
+In order to run this project you need:
+
+- Install Ruby
+- Install Rails
+- Install Postgresql
+
+### Setup
+
+Clone this repository to your desired folder:
+
+```sh
+  cd hello-rails-front-end
+  git clone https://github.com/TayyabKM/hello-react-front-end.git
+```
+
+### Install
+
+Install this project with:
+
+```sh
+  cd hello-rails-react
+  npx create-react-app
+```
+```sh
+  gem install --no-document rubocop -v '>= 1.0, < 2.0'
+```
+```sh
+  npm install --save-dev stylelint@13.x stylelint-scss@3.x stylelint-config-standard@21.x stylelint-csstree-validator@1.x
+```
+
+### Usage
+
+To run the project, execute the following command:
+
+```sh
+  yarn build
+```
+
+```sh
+  rails server
+```
+
+### Run tests
+
+To run tests, run the following command:
+
+```sh
+  rubocop --color
+```
+```sh
+  npx stylelint "**/*.{css,scss}"
+```
+```sh
+  rspec
+```

```

Comment:
- [ ] There is conflicting information here about how to install a react app. We do not use the same commands used to install a Rails app for a React app. Kindly add the correct information.